### DISCUSSION TITLE
Issue with payload in generate:extendRoutes

### DISCUSSION THREAD

**Discussion by anandtoshniwal** [nuxt/nuxt#9455] [Questions (Nuxt 2)] [answered]
Am new to Nuxt, payload works with npm generate but when I run dev build [npm run dev] payload is undefined is it expected behavior? or am I missing something here which causing this issue?

**Comment by danielroe** [comment, marked as answer]
You're right, payload isn't defined in dev mode. You should check if the payload exists, and if not, fetch the data freshly.

If you are sharing payload data across a site, you could do this in a plugin rather than in the component where you're using the payload.

**Reply by anandtoshniwal** [reply]
Creating routes in a module file  sharing payload through it in the component and it works in generate mode and not in dev
